layers/core/layer.py

```diff
@@ -0,0 +1,95 @@
+import json
+from layers.core.exceptions import UninitializedLayer, BadType, BadInput, \
+    handler
+from layers.core.layerobj import _LayerObj
+
+
+class Layer:
+    def __init__(self, init_dict={}, strict=True):
+        """
+             Initialization - create a new Layer object
+             :param init_dict: Optionally provide base Layer json or string
+                data on initialization
+         """
+        self.__layer = None
+        self.strict = strict
+        self.load_input(init_dict)
+
+    @property
+    def layer(self):
+        if self.__layer is not None:
+            return self.__layer
+        return "No Layer Loaded Yet!"
+
+    def load_input(self, init_dict):
+        """
+            loads input from a string or dictionary format
+            :param init_dit: the string or dictionary representing the layer
+                data to be loaded
+        """
+        if isinstance(init_dict, str):
+            self.data = json.loads(init_dict)
+        else:
+            self.data = init_dict
+        if self.data != {}:
+            self._build()
+
+    def load_file(self, filename):
+        """
+             loads input from a layer file specified by filename
+             :param filename: the target filename to load from
+        """
+        with open(filename, 'r') as fio:
+            raw = fio.read()
+            self.data = json.load(raw)
```

Comment:
This needs to be json.loads, not json.load. Load reads a filestream, loads reads a string.